### PR TITLE
Integration testing improvements

### PR DESCRIPTION
**Pull request by andregasser**
Improved test structure, added a few tests for MediaMethods and StatusMethods.

### CODE REVIEW

File: config/detekt/detekt.yml [hunk 1]
```diff
@@ -426,13 +426,14 @@ naming:
...
-    maximumFunctionNameLength: 50
+    maximumFunctionNameLength: 120
   FunctionMinLength:
     active: true
     minimumFunctionNameLength: 3
   FunctionNaming:
     active: true
-    excludes: ['**/test/**', '**/androidTest/**', '**/commonTest/**', '**/jvmTest/**', '**/jsTest/**', '**/iosTest/**']
+    excludes: ['**/test/**', '**/integrationTest/**']
+    #functionPattern: '[a-z`][a-zA-Z0-9`\s]*'
```
Comment (bocops) [inline config/detekt/detekt.yml:436]: Do we need to increase allowed function name length if we exclude integration test files from that check?

File: bigbone/src/integrationTest/kotlin/social/bigbone/TestHelpers.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * class (e.g. V402TestHelpers.kt) in the appropriate integration test directory.
+ */
+object TestHelpers {
+    /**
+     * Uploads an image file using the POST /api/v1/media REST call.
+     * @param filename the filename. Must be located in the "resources" directory".
+     * @param mediaType the media type of the file (e.g. "image/jpg", ...).
+     * @param client an instance of MastodonClient. This instance is used to conduct the upload.
+     * @return the media attachment information.
+     */
```
Comment (bocops) [inline bigbone/src/integrationTest/kotlin/social/bigbone/TestHelpers.kt:21]: How strict do we want to be about documentation for our test classes? This is the only method documentation in the class - but I guess the rest is pretty straightforward.

**Review by bocops — APPROVED** [review]
Looks good! I added two comments about things I'm not completely sure about, but would approve even without further changes.

File: config/detekt/detekt.yml (hunk 1 above)
Comment (andregasser) [reply config/detekt/detekt.yml:436]: I have disabled the `FunctionMaxLength` check, as it does not provide any options to exclude test sources.

File: bigbone/src/integrationTest/kotlin/social/bigbone/TestHelpers.kt (hunk 2 above)
Comment (andregasser) [reply bigbone/src/integrationTest/kotlin/social/bigbone/TestHelpers.kt:21]: Yes, well, my vote basically is for having the code written in a way that it does not need KDoc. And in this case, it is pretty obvous what the method does. So I can remove it. I wouldn't be strict with KDoc on test code.....

**Pull request merged**